lib/ls-utils.ts

```diff
@@ -41,6 +41,8 @@ async function iterateFiles(
  * directory sequence.
  */
 function parseLsOutput(output: string): DiscoveredDirectory {
+  const outputLines = output.trim().split("\n");
+  const isRecursiveOutput = outputLines[0].endsWith(":");
```

Comment:
won't this mistaken a file name that ends with ```:``` as a false indication of recursive scan?

Reply:
It should be fine for any files which end in `:` as the first line should always be `./` so:
```
 > ls -1ap
./
../
.:
.dockerenv
:
app/
bin/
dev/
etc/
foo:
home/
```
and (here the recursive case):
```
> ls -1apR
.:
./
../
.:
.dockerenv
:
app/
bin/
dev/
etc/
foo:
home/
```